stl/inc/xutility

```diff
@@ -4679,6 +4679,42 @@ template <class _Elem>
 _INLINE_VAR constexpr bool _Equal_memcmp_is_safe_helper<_Elem, _Elem, equal_to<_Elem>> =
     _Equal_memcmp_is_safe_helper<_Elem, _Elem, equal_to<>>;
 
+#ifdef __cpp_lib_concepts
+#ifdef __cpp_lib_byte
+// allow memcmping std::byte with _RANGES equal_to
+template <>
+inline constexpr bool _Equal_memcmp_is_safe_helper<byte, byte, _RANGES equal_to> = true;
+#endif // __cpp_lib_byte
+
+template <class _Pr>
+concept _Equal_to_comparator = is_same_v<_Pr, equal_to<>> || is_same_v<_Pr, _RANGES equal_to>;
+
+// allow memcmping same-size integral non-bool non-volatile bitwise types using equal_to
+// being integral is a precondition of _Value_equality_is_bitwise_equality
+// clang-format off
+template <class _Elem1, class _Elem2, _Equal_to_comparator _Pr>
+    requires (sizeof(_Elem1) == sizeof(_Elem2))
+          && _Is_nonbool_integral<_Elem1> && (!is_volatile_v<_Elem1>)
+          && _Is_nonbool_integral<_Elem2> && (!is_volatile_v<_Elem2>)
+inline constexpr bool _Equal_memcmp_is_safe_helper<_Elem1, _Elem2, _Pr> =
+    _Value_equality_is_bitwise_equality<_Elem1, _Elem2>;
+// clang-format on
+
+template <class _Iter>
+concept _Bitwise_comparable_iterator =
+    (is_pointer_v<_Iter> && !is_volatile_v<remove_pointer_t<_Iter>>) || contiguous_iterator<_Iter>;

```

Comment:
`is_pointer_v<_Iter>` implies `contiguous_iterator<_Iter>`, so we need not check both, and we probably want the "but not iterators with `volatile`-qualified element type" check to apply to general iterators as well.
```suggestion
    contiguous_iterator<_Iter> && !is_volatile_v<remove_reference_t<iter_reference_t<_Iter>>>;
```